testsuite/client-old-server/src/test/java/org/wildfly/client/old/server/ClientAgainstOldServerTestCase.java

```diff
@@ -0,0 +1,137 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.inject.Inject;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.standalone.ServerDeploymentManager;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.client.old.server.util.LargeDeploymentFile;
+import org.wildfly.client.old.server.util.OldVersionTestParameter;
+import org.wildfly.client.old.server.util.OldVersionTestRunner;
+import org.wildfly.client.old.server.util.Version;
+import org.wildfly.core.testrunner.ManagementClient;
+
+
+/**
+ * The OldVersionTestRunner generates the file, and its size is controlled by -Djboss.test.client.old.server.size={@code <bytes>}.
+ * The default is 1GB.
+ *
+ * @author Kabir Khan
+ */
+@RunWith(OldVersionTestRunner.class)
+public class ClientAgainstOldServerTestCase {
+
+    // Max time to wait for some action to complete, in s
+    private static final int TIMEOUT = TimeoutUtil.adjust(120);
+
+    private final Version.AsVersion version;
+
+    @Inject
+    protected ManagementClient managementClient;
+
+    @Inject
+    protected LargeDeploymentFile testDeploymentFile;
+
+    public ClientAgainstOldServerTestCase(OldVersionTestParameter param) {
+        this.version = param.getAsVersion();
+    }
+
+    @OldVersionTestRunner.OldVersionParameter
+    public static List<OldVersionTestParameter> parameters(){
+        return OldVersionTestParameter.setupVersions();
+    }
+
+    @Test
+    public void testVersion() throws Exception {
+        checkVersion();
+    }
+
+    @Test
+    public void testDeployment() throws Exception {
+        final ModelControllerClient client = managementClient.getControllerClient();
+        final ServerDeploymentManager manager = ServerDeploymentManager.Factory.create(client);
+
+        //Deploy
+        try (final InputStream is = testDeploymentFile.getInputStream()){
+            Future<?> future = manager.execute(
+                    manager.newDeploymentPlan().add(testDeploymentFile.getName(), is).deploy(testDeploymentFile.getName()).build());
+            awaitDeploymentExecution(future);
+        }
+
+        //Check there
+        ModelNode findDeployment = Util.createEmptyOperation(READ_RESOURCE_OPERATION,
+                PathAddress.pathAddress(DEPLOYMENT, testDeploymentFile.getName()));
+        Assert.assertTrue(
+                managementClient.executeForResult(findDeployment).isDefined());
+
+        //Undeploy
+        Future<?> future = manager.execute(manager.newDeploymentPlan().undeploy(testDeploymentFile.getName()).remove(testDeploymentFile.getName()).build());
```

Comment:
If you remove the ".deploy(testDeploymentFile.getName())" then the ".undeploy(testDeploymentFile.getName())" goes too.